mbuild/compound.py

```diff
@@ -315,6 +327,34 @@ def particles_by_element(self, element):
         for particle in self.particles():
             if particle.element == element:
                 yield particle
+    
+    @property
+    def mass(self):
+        if self._mass:
+            return self._mass
+        else:
+            return sum([self._particle_mass(p) for p in self.particles()]) 
+    
+    def _particle_mass(self, particle):
+        if particle._mass:
+            return particle._mass
+        else:
+            if particle.element:
+                return particle.element.mass
+            else:
+                return 0
+    
+    @mass.setter
+    def mass(self, value):
+        if value < 0:
+            raise ValueError(
+                    "Cannot set a mass value less than zero")
+        if self.children:
+            raise MBuildError(
+                "Cannot set the mass of a Compound containing "
+                "children compounds")
+        else:
+            self._mass = value
```

Comment:
We probably need a typecast to float right?

`self._mass = float(value)`

Also, what would happen if a user passed in a numpy array of floats? Im pretty sure that `if value < 0` would return a numpy mask array of booleans, so i think it would still catch above, but we should be sure.

Reply:
note: the `if value < 0` is referring to the first conditional in the mass setter

Reply:
Good points, I'll work on making this more robust

Reply:
> We probably need a typecast to float right?
> 
> `self._mass = float(value)`
> 
> Also, what would happen if a user passed in a numpy array of floats? Im pretty sure that `if value < 0` would return a numpy mask array of booleans, so i think it would still catch above, but we should be sure.

Good point, there should be some type checking/handling for `value`, but I'm not sure how passing in an array of floats (or anything containing more than one value) would work. Do you have an example?

If someone has a system of N particles, and they want to set/change the mass for all of them, then I think they'd have to do that one at a time in a for loop rather than passing in a list/array of values to the mass setter. Maybe I'm confused and that's not what you meant?

Reply:
The latest changes include handling of `value` if an iterable is passed in. If it contains more than one element an error is given, if not it continues and type casts the value to a `float`